lib/clients/ask-smapi-client/index.js

```diff
@@ -0,0 +1,83 @@
+const { CustomSmapiClientBuilder } = require('ask-smapi-sdk');
+
+const AppConfig = require('@src/model/app-config');
+const AuthorizationController = require('@src/controllers/authorization-controller');
+const DynamicConfig = require('@src/utils/dynamic-config');
+const Messenger = require('@src/view/messenger');
+const jsonView = require('@src/view/json-view');
+
+const _stringifyHeaders = (item) => {
+    let headers = {};
+    item.headers.forEach(h => {
+        headers[h.key] = h.value;
+    });
+    headers = JSON.stringify(headers);
+    return { ...item, headers };
+};
+
+/**
+ * @param {Object}request Request object
+ * @param {Array<{key, value}>} request.headers Array of headers
+ * @param {string} request.method HTTP method
+ * @param {string} request.url HTTP url
+ * @param {Object} request.body HTTP body
+ */
+const logRequest = (req) => {
+    Messenger.getInstance().info('REQUEST:');
+    Messenger.getInstance().info(jsonView.toString(_stringifyHeaders(req)));
+};
+
+/**
+ * @param {Object} response Response object
+ * @param {Array<{key, value}>} response.headers Array of headers
+ * @param {string} response.statusCode HTTP status code
+ * @param {Object} response.body HTTP body
+ */
+const logResponse = (res) => {
+    Messenger.getInstance().info('RESPONSE:');
+    const message = _stringifyHeaders(res);
+    const requestId = res.headers.find(h => h.key === 'x-amzn-requestid');
+    message['request-id'] = requestId ? requestId.value : null;
+    Messenger.getInstance().info(jsonView.toString(message));
+};
+
+/**
+ * @param {Object} config Configuration object
+ * @param {string} config.profile ASK CLI profile
+ * @param {boolean} config.debug debug flag
+ */
+const makeSmapiClient = ({ profile, debug = false }) => {
+    new AppConfig();
+    const authorizationController = new AuthorizationController({
+        auth_client_type: 'LWA',
+        doDebug: debug
+    });
+
+    const refreshTokenConfig = {
+        clientId: authorizationController.oauthClient.config.clientId,
+        clientSecret: authorizationController.oauthClient.config.clientConfirmation,
+        refreshToken: AppConfig.getInstance().getToken(profile).refresh_token
```

Comment:
We still need to support the two env var mentioned here https://github.com/alexa/ask-cli/blob/develop/lib/controllers/authorization-controller/index.js#L50-L51

We mentioned the usage in our doc https://github.com/alexa/ask-cli/blob/develop/docs/concepts/CI-CD.md about the existence of those two env var. However, smapi-sdk doesn't support the two env var yet. Should this work be on smapi-sdk?